### DISCUSSION TITLE
Usage for an iPhone app?

### DISCUSSION THREAD

**Discussion by fivestones** [vlcn-io/cr-sqlite#149] [Q&A] [answered]
This looks amazing.

Is there any obvious way of making use of this for an iPhone app? Would I be limited to some kind of progressive web app, or maybe making something in javascript through reactive native? Or is there some other straightforward way to use this, e.g., in a swift app?

**Comment by tantaman** [comment]
Yep, you can compile and run the extension on an iPhone as well and access it from Swift.

I've talked to some people who have done this although I haven't done it yet myself. Let me see if I can get them to chime in here on how this was done.

**Comment by fivestones** [comment]
That sounds great--hopefully they can share some of the details here. Thanks!

**Comment by fivestones** [comment]
Over the past couple of days I've been trying to figure out how to load an extension for sqlite in swift, but no luck so far. I did find some info about packaging a dylib in an xcframework to be able to use it in an iOS app, but I haven't gotten very far on this yet.

**Comment by tantaman** [comment]
I'll spend some time today trying to figure it out.

**Comment by tantaman** [comment]
Looks like you could compile the sqlite library with the extension statically linked -- https://stackoverflow.com/questions/6283496/xcode-sqlite-extension-for-use-on-iphone
&
https://stackoverflow.com/questions/2303140/how-to-link-a-sqlite-extension-source-file-into-xcode-for-iphone

this is how we do the WASM build currently so shouldn't be too big a leap to get the same thing working for iOS.

I'm on vacation next week but might be able to carve out a few hours.

**Comment by tantaman** [comment]
Made some progress on this here: https://github.com/vlcn-io/cr-sqlite/pull/276 & related issue: https://github.com/vlcn-io/cr-sqlite/issues/275

**Comment by tantaman** [comment]
Expo has an ios build they've integrated into https://expo.dev (https://github.com/expo/expo/pull/23791)

I'll start publishing ios builds in the next release.

**Comment by tantaman** [comment]
the v0.15 preview release has ios builds -- https://github.com/vlcn-io/cr-sqlite/releases/tag/v0.15.0-next.1

[crsqlite-ios-dylib.xcframework.tar.gz](https://github.com/vlcn-io/cr-sqlite/releases/download/v0.15.0-next.1/crsqlite-ios-dylib.xcframework.tar.gz)

**Note that v0.15 will not be backwards compatible with v0.14.**

v0.15 renamed all the API methods to be consistently snake cased. E.g., `siteid` -> `site_id`, `nextdbversion` -> `next_db_version` as well as added support for re-inserting rows with the same primary key as previously deleted rows. The latter changed the metadata format and merge logic quite a bit.

**Comment by tantaman** [comment, marked as answer]
All current and future releases now contain ios builds in the `xcframework.tar.gz` artifact.
